plugins/hypervisors/xenserver/src/com/cloud/hypervisor/xenserver/resource/CitrixResourceBase.java

```diff
@@ -625,15 +627,20 @@ protected ExecutionResult cleanupNetworkElementCommand(final IpAssocCommand cmd)
 
                 // there is only one ip in this public vlan and removing it, so
                 // remove the nic
-                if (ipsCount == 1 && !ip.isAdd()) {
-                    removeVif = true;
+                if (lastIp != null && lastIp.equalsIgnoreCase("true") && !ip.isAdd()) {
```

Comment:
@ProjectMoon  
Updated to use org.apache.commons.lang.StringUtils
Can you please review it.